### PR TITLE
ImageMagickCommandFactory: only extract first layer for PDFs

### PR DESCRIPTION
**Pull request by dunn**
Should fix black images; see https://github.com/samvera/hydra-derivatives/issues/110 and https://github.com/samvera/hydra-derivatives/pull/120

### CODE REVIEW

**Comment by dunn** [comment]
Mostly for brainstorming cc @jcoyne

File: app/models/riiif/image.rb [hunk 1]
```diff
@@ -51,7 +51,10 @@ def render(args)
     def info
       @info ||= begin
                   result = info_service.call(id, file)
-                  ImageInformation.new(width: result[:width], height: result[:height])
+                  ImageInformation.new(
```
Comment (jcoyne) [inline app/models/riiif/image.rb]: I'd prefer we make `ImageInformation` have a dimensions attribute and add the format attribute.  Is there a reason we can't do that?

Comment (dunn) [reply app/models/riiif/image.rb]: Yeah, I think I was confused by the weird override of the `initialize` method.  Updated now though.

**Comment by dunn** [comment]
Getting a bunch of failures like this:
```
  17) Riiif::Image#render quality when bitonal is specified converts to bitonal
      Failure/Error:
        height, width, format = Riiif::CommandRunner.execute(
          "#{external_command} -format '%h %w %m' #{@path}[0]"
        ).split(' ')

      NoMethodError:
        undefined method `split' for nil:NilClass
      # ./app/services/riiif/image_magick_info_extractor.rb:15:in `extract'
      # ./app/models/riiif/file.rb:53:in `info'
      # ./app/models/riiif/image.rb:19:in `block (2 levels) in <class:Image>'
      # ./app/models/riiif/image.rb:18:in `block in <class:Image>'
      # ./app/models/riiif/image.rb:53:in `info'
      # ./app/models/riiif/image.rb:47:in `block in render'
      # ./app/models/riiif/image.rb:46:in `render'
      # ./spec/models/riiif/image_spec.rb:282:in `block (4 levels) in <top (required)>'
      # ./spec/models/riiif/image_spec.rb:332:in `block (5 levels) in <top (required)>'
```
There must be something buggy with the `CommandRunner`, because when I run the same command directly with `popen3` it works fine.

**Comment by dunn** [comment]
Why do we benchmark the commands, anyway?

**Comment by jcoyne** [comment]
Looks like there's a few tests to fix up.

**Comment by dunn** [comment]
@jcoyne yeah, that's the CommandRunner stuff that's confusing me.

**Comment by dunn** [comment]
If the benchmarking isn't required, we could just remove the CommandRunner entirely.

**Comment by dunn** [comment]
The bizarre thing is I set a `byebug if out.nil?` debug trigger in `#execute`; it's never tripped, but somehow the downstream method calling it gets a `nil` response.

I thought maybe a different `#execute` method was somehow getting called instead, but I changed everything to be explicitly `Riiif::CommandRunner.execute` and it still happens.

**Comment by dunn** [comment]
Messing around, I tried replacing `#execute` just with backticks and `#strip` and got encoding errors, but I don't know if that's indicative of anything.

File: spec/models/riiif/image_spec.rb [hunk 2]
```diff
@@ -19,11 +19,7 @@
     end
 
     it 'renders' do
-      expect(Riiif::CommandRunner).to receive(:execute)
-        .with("convert -quality 85 -sampling-factor 4:2:0 -strip #{filename} jpg:-")
-        .and_return('imagedata')
-
-      expect(subject.render('size' => 'full', format: 'jpg')).to eq 'imagedata'
+      subject.render('size' => 'full', format: 'jpg')
```
Comment (jcoyne) [inline spec/models/riiif/image_spec.rb]: This no longer tests anything. Can you restore the `Riiif::CommandRunner`?

File: spec/models/riiif/image_spec.rb [hunk 3]
```diff
@@ -69,20 +65,13 @@
   end
 
   describe '#render' do
-    before do
-      allow(Riiif::CommandRunner).to receive(:execute)
```
Comment (jcoyne) [inline spec/models/riiif/image_spec.rb]: All the errors are a result of this line being removed.

**Comment by jcoyne** [comment]
Thanks, this looks great!

**Pull request merged**